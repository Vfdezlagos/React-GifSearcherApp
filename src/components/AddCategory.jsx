import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState("");

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        // Validar que el input value no este vacio
        // const isValid = !!inputValue.trim();
        // if (isValid) setCategories([inputValue, ...categories]);

        const value = inputValue.trim();

        if (value.length <= 1) return;

        // setCategories((categories) => [inputValue, ...categories]);
        onNewCategory(value);
        setInputValue("");
    };

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func,
    // categories: PropTypes.array.isRequired,
};
