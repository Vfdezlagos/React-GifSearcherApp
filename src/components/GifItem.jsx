import PropTypes from "prop-types";

export const GifItem = ({ id, title, url }) => {
    return (
        <div className="card" key={id}>
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    );
};

GifItem.propTypes = {
    // image: PropTypes.object,
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};
