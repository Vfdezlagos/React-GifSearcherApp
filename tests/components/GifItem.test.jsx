/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem component", () => {
    const id = "idprueba";
    const title = "titlePrueba";
    const url = "http://urlprueba.com/";

    const image = { id, title, url };

    test("Debe coincidir con el snapshot", () => {
        const { container } = render(<GifItem key={id} {...image} />);
        expect(container).toMatchSnapshot();
    });

    test("Debe de mostrar la imagen con el url y el alt indicado", () => {
        render(<GifItem key={id} {...image} />);
        const { src, alt } = screen.getByRole("img");
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test("Debe de mostrar el titulo en el componente", () => {
        render(<GifItem key={id} {...image} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});
