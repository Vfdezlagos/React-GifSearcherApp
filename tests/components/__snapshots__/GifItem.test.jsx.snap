// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem component Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titlePrueba"
      src="http://urlprueba.com/"
    />
    <p>
      titlePrueba
    </p>
  </div>
</div>
`;
